internal/configprovider/config_source_provider.go

```diff
@@ -62,7 +62,16 @@ func (c *configSourceConfigMapProvider) Retrieve(
 	if err != nil {
 		return config.Retrieved{}, err
 	}
-	c.wrappedRetrieved = wr
+
+	existingMap, _ := c.wrappedRetrieved.AsMap()
```

Comment:
Even though hardcoded as nil atm, it's probably best to handle and expose the error.

Reply:
Updated